File: robocode-tankroyale-server/src/main/java/dev/robocode/tankroyale/server/server/ConnListener.java
package dev.robocode.tankroyale.server.server;

import java.util.Collection;

import dev.robocode.tankroyale.schema.*;
import org.java_websocket.WebSocket;


@SuppressWarnings({"WeakerAccess", "EmptyMethod", "unused"})
public interface ConnListener {

	void onException(Exception exception);

	void onBotJoined(WebSocket conn, BotHandshake handshake);

	void onBotLeft(WebSocket conn, BotHandshake handshake);

	void onObserverJoined(WebSocket conn, ObserverHandshake handshake);

	void onObserverLeft(WebSocket conn, ObserverHandshake handshake);

	void onControllerJoined(WebSocket conn, ControllerHandshake handshake);

	void onControllerLeft(WebSocket conn, ControllerHandshake handshake);

	void onBotReady(WebSocket conn);

	void onBotIntent(WebSocket conn, BotIntent intent);

	void onStartGame(GameSetup gameSetup, Collection<BotAddress> botAddresses);

	void onAbortGame();

	void onPauseGame();

	void onResumeGame();

	void onChangeTps(int tps);
}

File: robocode-tankroyale-server-kotlin/src/main/kotlin/dev/robocode/tankroyale/server/conn/ConnListener.kt
package dev.robocode.tankroyale.server.dev.robocode.tankroyale.server.conn

import dev.robocode.tankroyale.schema.*
import org.java_websocket.WebSocket

/** Connection listener interface. */
interface ConnListener {
    fun onException(exception: Exception)
    fun onBotJoined(conn: WebSocket, handshake: BotHandshake)
    fun onBotLeft(conn: WebSocket, handshake: BotHandshake)
    fun onObserverJoined(conn: WebSocket, handshake: ObserverHandshake)
    fun onObserverLeft(conn: WebSocket, handshake: ObserverHandshake)
    fun onControllerJoined(conn: WebSocket, handshake: ControllerHandshake)
    fun onControllerLeft(conn: WebSocket, handshake: ControllerHandshake)
    fun onBotReady(conn: WebSocket)
    fun onBotIntent(conn: WebSocket, intent: BotIntent)
    fun onStartGame(gameSetup: GameSetup, botAddresses: Collection<BotAddress>)
    fun onAbortGame()
    fun onPauseGame()
    fun onResumeGame()
    fun onChangeTps(tps: Int)
}
